go/merkletree/merkle_verifier.go

```diff
@@ -46,15 +46,16 @@ func (m MerkleVerifier) VerifyInclusionProof(leafIndex, treeSize int64, proof []
 }
 
 // RootFromInclusionProof calculates the expected tree root given the proof and leaf.
+// leafIndex starts at 0. treeSize starts at 1.
 func (m MerkleVerifier) RootFromInclusionProof(leafIndex, treeSize int64, proof [][]byte, leaf []byte) ([]byte, error) {
-	if leafIndex > treeSize {
+	if leafIndex >= treeSize {
 		return nil, fmt.Errorf("leafIndex %d > treeSize %d", leafIndex, treeSize)
 	}
-	if leafIndex == 0 {
-		return nil, errors.New("leafIndex is zero")
+	if leafIndex < 0 || treeSize < 0 {
```

Comment:
I was following the [python](https://github.com/google/certificate-transparency/blob/master/python/ct/crypto/merkle.py#L482) code.